### PR TITLE
 [XPU] Add layernorm_relu pass and kernel

### PR DESCRIPTION
**Pull request by gitliuyf**
### PR Category
Performance Optimization

### PR Types
New features

### Description
新增融合算子，将layer_norm和relu进行了融合。在xpu平台上，经过实测，性能有提升

### CODE REVIEW

**Comment by newway** [comment]
lgtm

File: paddle/phi/kernels/fusion/xpu/layer_norm_relu_xpu_kernel.cc [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  xpu::ctx_guard RAII_GUARD(ctx.x_context());
+
+  // scale
+  const float* scale_data_fp32 = nullptr;
+  const auto* scale_ptr = scale.get_ptr();
+  if (scale_ptr == nullptr) {
+    // no scale, do nothing
+  } else if (scale_ptr->dtype() ==
+             phi::CppTypeToDataType<phi::dtype::float16>::Type()) {
```
Comment (zhupengyang) [inline paddle/phi/kernels/fusion/xpu/layer_norm_relu_xpu_kernel.cc]: phi::DataType::FLOAT16

File: paddle/phi/kernels/fusion/xpu/layer_norm_relu_xpu_kernel.cc [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    scale_data_fp32 = scale_ptr->data<float>();
+  }
+
+  // bias
+  const float* bias_data_fp32 = nullptr;
+  const auto* bias_ptr = bias.get_ptr();
+  if (bias_ptr == nullptr) {
+    // no bias, do nothing
+  } else if (bias_ptr->dtype() ==
+             phi::CppTypeToDataType<phi::dtype::float16>::Type()) {
```
Comment (zhupengyang) [inline paddle/phi/kernels/fusion/xpu/layer_norm_relu_xpu_kernel.cc]: phi::DataType::FLOAT16

File: paddle/phi/ops/yaml/fused_ops.yaml [hunk 3]
```diff
@@ -590,6 +590,15 @@
...
 
+- op : layer_norm_relu_xpu
+  args : (Tensor x, Tensor scale, Tensor bias, int begin_norm_axis, float epsilon = 1e-5)
+  output : Tensor(out)
+  infer_meta :
+    func : LayerNormalizeReluXPUInferMeta
+  kernel :
+    func : layer_norm_relu_xpu
+    data_type : x
+
```
Comment (zhupengyang) [inline paddle/phi/ops/yaml/fused_ops.yaml]: optional : sacle, bias

Comment (gitliuyf) [reply paddle/phi/ops/yaml/fused_ops.yaml]: done

File: paddle/phi/kernels/fusion/xpu/layer_norm_relu_xpu_kernel.cc (hunk 2 above)
Comment (gitliuyf) [reply paddle/phi/kernels/fusion/xpu/layer_norm_relu_xpu_kernel.cc]: done

File: paddle/phi/kernels/fusion/xpu/layer_norm_relu_xpu_kernel.cc (hunk 1 above)
Comment (gitliuyf) [reply paddle/phi/kernels/fusion/xpu/layer_norm_relu_xpu_kernel.cc]: done

File: test/ir/inference/test_xpu_layer_norm_relu_pass.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        self.run_and_statis(
+            quant=False,
+            max_examples=25,
+            passes=["layer_norm_relu_xpu_fuse_pass"],
+        )
+
+
+if __name__ == "__main__":
+    np.random.seed(200)
+    unittest.main()
```
Comment (cqulilujia) [inline test/ir/inference/test_xpu_layer_norm_relu_pass.py]: from paddle.framework import core
if core.get_xpu_device_version(0) == core.XPUVersion.XPU2:
    unittest.main()

Comment (gitliuyf) [reply test/ir/inference/test_xpu_layer_norm_relu_pass.py]: done

File: test/ir/inference/test_xpu_layer_norm_relu_pass.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+import unittest
+
+import hypothesis.strategies as st
+import numpy as np
+from auto_scan_test import PassAutoScanTest
+from program_config import OpConfig, ProgramConfig, TensorConfig
+
+from paddle.framework import core
+
+
```
Comment (cqulilujia) [inline test/ir/inference/test_xpu_layer_norm_relu_pass.py:24]: @unittest.skipIf(
    not core.get_xpu_device_version(0) == core.XPUVersion.XPU2,
    "XpuLayerNormReluFuse only support XPU2",
)

Comment (gitliuyf) [reply test/ir/inference/test_xpu_layer_norm_relu_pass.py:24]: done

**Review by wanghuancoder — APPROVED** [review]

**Pull request merged**